src/java.net.http/share/classes/jdk/internal/net/http/RequestPublishers.java

```diff
@@ -487,7 +483,7 @@ public ByteBuffer next() {
                 }
             } catch (IOException ioe) {
                 terminated = true;
-                throw new UncheckedIOException(ioe);
+                throw new IOException(ioe);
```

Comment:
We don't convert `IOE`s to `UncheckedIOE`s anymore. This effectively makes the `HttpClientImpl::send` exception translation logic branching on `if (throwable instanceof IOException)` to kick in.